StorageUtility/Linux/StorageUtility.cpp

```diff
@@ -0,0 +1,282 @@
+                if ( eOnErrorBehavior::Continue == OnErrorBehavior ) { continue; }
+                SetupDiDestroyDeviceInfoList( devInfoHandle );
+                return( eErrorCode::Unknown );
+            }
+        }
+
+        eErrorCode errorCode;
+        errorCode = eErrorCode::Unknown;
+        Callback.Function(Callback.Data, devInfoHandle, devInfoData, devInterfaceData, pDevDetailData, bufferSize, errorCode);
+
+        HeapFree(GetProcessHeap(), HEAP_NO_SERIALIZE, pDevDetailData);
+    }
+
+    SetupDiDestroyDeviceInfoList( devInfoHandle );
+
+    return( eErrorCode::None );
+}*/
+
+eErrorCode GetStorageAdapterProperty( HANDLE Handle, sStorageAdapterProperty& AdapterProperty )
+{
+   /* STORAGE_DESCRIPTOR_HEADER storageDescHeader;
+
+    STORAGE_PROPERTY_QUERY StorageProperty;
+    StorageProperty.PropertyId = StorageAdapterProperty;
+    StorageProperty.QueryType = PropertyStandardQuery;
+
+    DWORD bytesReturned;
+    DWORD ret =
+        DeviceIoControl(
+                        Handle,
+                        IOCTL_STORAGE_QUERY_PROPERTY,
+                        &StorageProperty,
+                        sizeof(STORAGE_PROPERTY_QUERY),
+                        &storageDescHeader,
+                        sizeof(STORAGE_DESCRIPTOR_HEADER),
+                        &bytesReturned,
+                        NULL
+                        );
+    if ( 0 == ret )
+    {
+        return( eErrorCode::Io );
+    }
+
+    if ( 0 == storageDescHeader.Size )
+    {
+        return( eErrorCode::Io );   //TODO: replace with more descriptive error
+    }
+
+    PSTORAGE_ADAPTER_DESCRIPTOR pstorageAdapterDesc = NULL;
+    pstorageAdapterDesc = (PSTORAGE_ADAPTER_DESCRIPTOR)HeapAlloc(GetProcessHeap(), HEAP_GENERATE_EXCEPTIONS, storageDescHeader.Size);
+
+    ret =
+        DeviceIoControl(
+                        Handle,
+                        IOCTL_STORAGE_QUERY_PROPERTY,
+                        &StorageProperty,
+                        sizeof(STORAGE_PROPERTY_QUERY),
+                        pstorageAdapterDesc,
+                        storageDescHeader.Size,
+                        &bytesReturned,
+                        NULL
+                        );
+    if (0 == ret)
+    {
+        HeapFree(GetProcessHeap(), HEAP_NO_SERIALIZE, pstorageAdapterDesc);
+        return( eErrorCode::Io );
+    }
+
+    AdapterProperty.BusType = pstorageAdapterDesc->BusType;
+    //iAdapterProperty.SrbType = pstorageAdapterDesc->SrbType;  TODO check this
+    AdapterProperty.AlignmentMask = pstorageAdapterDesc->AlignmentMask;
+
+    HeapFree(GetProcessHeap(), HEAP_NO_SERIALIZE, pstorageAdapterDesc);
+*/
+    return( eErrorCode::None );
+}
+
+void CloseDeviceHandle(HANDLE& Handle)
+{
+//    if (FALSE == CloseHandle(Handle))
+//    {
+//        //! TODO
+//        //fprintf(stderr, "\nCloseDeviceHandle was not successful. Error Code: %d", GetLastError());
+//    }
+//    Handle = INVALID_HANDLE_VALUE;
+}
+
+bool IsAtaDeviceBus(udev_device* dev)

```

Comment:
Coding convention: the first character in name variables is written in capital like this: sStorageAdapterProperty StorageDeviceProperty .